lib/vdc/DCBOV.cpp

```diff
@@ -0,0 +1,317 @@
+
+    vector<string> names;
+    for (itr = _dataVarsMap.begin(); itr != _dataVarsMap.end(); ++itr) { names.push_back(itr->first); }
+    return (names);
+}
+
+std::vector<string> DCBOV::getCoordVarNames() const
+{
+    map<string, DC::CoordVar>::const_iterator itr;
+
+    vector<string> names;
+    for (itr = _coordVarsMap.begin(); itr != _coordVarsMap.end(); ++itr) { names.push_back(itr->first); }
+    return (names);
+}
+
+bool DCBOV::getAtt(string varname, string attname, vector<double> &values) const { return false; }
+
+bool DCBOV::getAtt(string varname, string attname, vector<long> &values) const { return false; }
+
+bool DCBOV::getAtt(string varname, string attname, string &values) const { return false; }
+
+std::vector<string> DCBOV::getAttNames(string varname) const { return {}; }
+
+DC::XType DCBOV::getAttType(string varname, string attname) const { return INVALID; }
+
+int DCBOV::getDimLensAtLevel(string varname, int, std::vector<size_t> &dims_at_level, std::vector<size_t> &bs_at_level) const
+{
+    dims_at_level.clear();
+    bs_at_level.clear();
+
+    bool ok = GetVarDimLens(varname, true, dims_at_level);
+    if (!ok) {
+        SetErrMsg("Undefined variable name : %s", varname.c_str());
+        return (-1);
+    }
+
+    // Never blocked
+    //
+    bs_at_level = vector<size_t>(dims_at_level.size(), 1);
+
+    return (0);
+}
+
+int DCBOV::openVariableRead(size_t ts, string varname)
+{
+    FileTable::FileObject *f = new FileTable::FileObject(ts, varname, 0, 0, 0);
+    return (_fileTable.AddEntry(f));
+}
+
+int DCBOV::closeVariable(int fd)
+{
+    DC::FileTable::FileObject *w = _fileTable.GetEntry(fd);
+
+    if (!w) {
+        SetErrMsg("Invalid file descriptor : %d", fd);
+        return (-1);
+    }
+
+    _fileTable.RemoveEntry(fd);
+    if (w != nullptr) delete w;
+
+    return 0;
+}
+
+template<class T> int DCBOV::_readRegionTemplate(int fd, const vector<size_t> &min, const vector<size_t> &max, T *region)
+{
+    FileTable::FileObject *w = (FileTable::FileObject *)_fileTable.GetEntry(fd);
+    if (!w) {
+        SetErrMsg("Invalid file descriptor : %d", fd);
+        return (-1);
+    }
+
+    std::string varname = w->GetVarname();
+    size_t      ts = w->GetTS();
+
+    std::array<std::string, 3> spatialDims = _bovCollection->GetSpatialDimensions();
+    std::array<size_t, 3>      dataSize = _bovCollection->GetDataSize();
+    std::array<double, 3>      origin = _bovCollection->GetBrickOrigin();
+    std::array<double, 3>      brickSize = _bovCollection->GetBrickSize();
+
+    // Return spatial coordinate variable values
+    for (int dim = 0; dim < spatialDims.size(); dim++) {
+        if (varname == spatialDims[dim]) {
+            double increment = brickSize[dim] / (dataSize[dim] - 1);
+            double start = origin[dim] + min[0] * increment;
+            size_t steps = max[0] - min[0];
+            for (int i = 0; i <= steps; i++) { region[i] = start + i * increment; }
```

Comment:
Please make this code clearer by encapsulating the search for a coord variable and the returning of a linear ramp into two separate functions.

Reply:
No problem.  Fixed.